agent/grpc-external/services/peerstream/stream_resources.go

```diff
@@ -331,13 +332,34 @@ func (s *Server) realHandleStream(streamReq HandleStreamRequest) error {
 
 			return nil
 
+		case err := <-errCh:
+			if err == io.EOF {
+				logger.Info("stream ended by peer")
+				return nil
+			}
+			grpcErr, ok := grpcstatus.FromError(err)
+			if ok {
+				// todo: return a sentinel error, don't string match.
```

Comment:
I think in gRPC cases since errors are transferred over the wire it might be hard to serialize sentinel go errors in a straightforward way. We could make the error strings sentinel consts so changes are reflected on both ends of the wire.

Reply:
I found a way that works pretty well by type matching on `Details()` that  i got from https://jbrandhorst.com/post/grpc-errors/